### PR TITLE
Added Dockerfile and devcontainer.json to support docker for vscode.

### PR DESCRIPTION
**Pull request by gkpln3**
IDK if someone here is interested in a Docker environment for the development environment, but i've made one.

### CODE REVIEW

**Comment by pebenito** [comment]
I'm not opposed to having these, but I won't be updating them. If over time they stop working and no one fixes it, they will be removed.

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+setuptools
+networkx
+Cython
+PyQt5==5.14
```
Comment (pebenito) [inline requirements.txt:4]: Is there a specific reason for pinning this version?

Comment (gkpln3) [reply requirements.txt:4]: `PyQt5` latest version (5.14.1) fails to install for some reason as mentioned on this stackoverflow thread:
https://stackoverflow.com/questions/59711301/install-pyqt5-5-14-1-on-linux

Comment (pebenito) [reply requirements.txt:4]: In that case, you should handle this in the Dockerfile. If there is a `requirements.txt` then it should reflect the same version dependencies as stated in the README.

**Review by pebenito — CHANGES_REQUESTED** [review]
Please move this into a `contrib` dir at the top level instead of a hidden `.devcontainer` dir. Then if other people want to contribute useful things like this, there is a proper place.

**Pull request closed without merging**